packages/core/src/components/renvooi/renvooi.tsx

```diff
@@ -1,29 +1,94 @@
-import { Component, ComponentInterface, Fragment, FunctionalComponent, Prop, h } from "@stencil/core";
+import {
+  Component,
+  ComponentInterface,
+  Fragment,
+  FunctionalComponent,
+  Prop,
+  h,
+  Event,
+  EventEmitter,
+} from "@stencil/core";
 
-import { RenvooiValue } from "./renvooi.interfaces";
+import {
+  RenvooiMarkFunction,
+  RenvooiMarkItemHighlightEvent,
+  RenvooiRenderMarkFunction,
+  RenvooiValue,
+} from "./renvooi.interfaces";
 
 interface RenvooiRenderProps {
   value: RenvooiValue;
+  mark: RenvooiRenderMarkFunction | undefined;
+  onDsoMarkItemHighlight: (text: string, elementRef: HTMLElement) => void | undefined;
 }
 
-const RenvooiRender: FunctionalComponent<RenvooiRenderProps> = ({ value }) => {
+interface RenderTextProps {
+  text: string;
+  mark: RenvooiRenderMarkFunction | undefined;
+  emitMarkItemHighlight: (text: string, ref: HTMLElement) => void | undefined;
+}
+
+const RenderText: FunctionalComponent<RenderTextProps> = ({ text, mark, emitMarkItemHighlight }) => {
+  if (!mark) {
+    return <>{text}</>;
+  }
+
+  const result = mark(text);
+
+  if (!result) {
+    return <>{text}</>;
+  }
+
+  return result.map((renvooiText) => {
```

Comment:
Als je deze nog wrapped in een fragment heeft deze Functional Component een mooie return type